### PR TITLE
Support hostnameIPv4 and hostnameIPv6

### PR DESCRIPTION
**Pull request by LinEvil**

### CODE REVIEW

**Comment by LinEvil** [comment]
- Support hostnameIPv4 and hostnameIPv6 
- Add EnableCompare
- Rename settings.jsonc to ~.example
- Send request to cloudflare without proxy

**Comment by database64128** [comment]
Can you make the use of proxy configurable too? In my use case, it's better to send the Cloudflare API request via proxy.

**Comment by LinEvil** [comment]
According to [MSFT docs](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.utility/invoke-webrequest?view=powershell-7.1#notes)
The request `Invoke-WebRequest` uses the proxy settings in the system environment variables by default.

Maybe an option to toggle it is helpful, I will do it later.

**Comment by database64128** [comment]
> Maybe an option to toggle it is helpful

That's what I meant. 😄

File: settings.jsonc.example [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    },
+    "IPv4": {
+        "Enable": true,
+        "Proxied": "false",
+        "Hostname": "example.com",
+        "CloudflareUri": "https://api.cloudflare.com/client/v4/zones/<ZONE_ID>/dns_records/<RECORD_ID>",
+        "AddressAPI": "https://api.ipify.org/"
+    },
+    // Resolve the hostname and compare with current IP to decide whether to send an update request.
+    "EnableCompare": true,
```
Comment (database64128) [inline settings.jsonc.example:23]: `Enable` -> `Enabled`

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,2 @@
+*.jsonc
+*.log
```
Comment (database64128) [inline .gitignore:2]: I prefer to have a new line at the end of a file.

File: settings.jsonc.example [hunk 3]
```diff
@@ -0,0 +1,26 @@
+{
+    // Leave empty to print to stdout.
+    "LogFilePath": "ddns.log",
+    // The interval for periodic checks in seconds, 0 for an one-shot run.
```
Comment (database64128) [inline settings.jsonc.example:4]: an -> a

**Review by database64128 — CHANGES_REQUESTED** [review]

File: settings.jsonc.example (hunk 1 above)
Comment (database64128) [reply settings.jsonc.example]: Sorry, GitHub seems to have messed up the position. `EnableCompare` is fine. My comment was targeting `IPv4.Enable` and `IPv6.Enable`.

**Comment by database64128** [comment]
Thank you for your work!

**Pull request merged**